airflow-core/src/airflow/serialization/serialized_objects.py

```diff
@@ -1906,7 +1906,11 @@ def serialize_dag(cls, dag: SdkDag) -> dict:
             serialized_dag["dag_dependencies"] = [x.__dict__ for x in sorted(dag_deps)]
             serialized_dag["task_group"] = TaskGroupSerialization.serialize_task_group(dag.task_group)
 
-            serialized_dag["deadline"] = dag.deadline.serialize_deadline_alert() if dag.deadline else None
+            serialized_dag["deadline"] = (
+                [deadline.serialize_deadline_alert() for deadline in dag.deadline]
+                if isinstance(dag.deadline, list)
+                else None
+            )
```

Comment:
Isn't is possible (as of now, but not after Ramit's suggestion)  for it not to be a list, but just a single value? So you'd need a third branch here?

Reply:
Same below in deserialize